include/zephyr/drivers/adc.h

```diff
@@ -146,6 +146,200 @@ struct adc_channel_cfg {
 #endif /* CONFIG_ADC_CONFIGURABLE_INPUTS */
 };
 
+/**
+ * @brief Get ADC channel configuration from a given devicetree node.
+ *
+ * This returns a static initializer for a <tt>struct adc_channel_cfg</tt>
+ * filled with data from a given devicetree node.
+ *
+ * Example devicetree fragment:
+ *
+ * @code{.dts}
+ * &adc0 {
+ *    #address-cells = <1>;
+ *    #size-cells = <0>;
+ *
+ *    channel@0 {
+ *        reg = <0>;
+ *        zephyr,gain = "ADC_GAIN_1_6";
+ *        zephyr,reference = "ADC_REF_INTERNAL";
+ *        zephyr,acquisition-time = <ADC_ACQ_TIME(ADC_ACQ_TIME_MICROSECONDS, 20)>;
+ *        bipolar;
+ *        zephyr,input-positive = <NRF_SAADC_AIN6>;
+ *        zephyr,input-negative = <NRF_SAADC_AIN7>;
+ *    };
+ * };
+ * @endcode
+ *
+ * Example usage:
+ *
+ * @code{.c}
+ * static const struct adc_channel_cfg ch0_cfg_dt =
+ *     ADC_CHANNEL_CFG_DT(DT_CHILD(DT_NODELABEL(adc0), channel_0));
+ *
+ * // Initializes 'ch0_cfg_dt' to:
+ * // {
+ * //     .channel_id = 0,
+ * //     .gain = ADC_GAIN_1_6,
+ * //     .reference = ADC_REF_INTERNAL,
+ * //     .acquisition_time = ADC_ACQ_TIME(ADC_ACQ_TIME_MICROSECONDS, 20),
+ * //     .differential = true,
+ * //     .input_positive = NRF_SAADC_AIN6,
+ * //     .input-negative = NRF_SAADC_AIN7,
+ * // }
+ * @endcode
+ *
+ * @param node_id Devicetree node identifier.
+ *
+ * @return Static initializer for an adc_channel_cfg structure.
+ */
+#define ADC_CHANNEL_CFG_DT(node_id) { \
```

Comment:
I wanted to avoid confusion, as this macro does not return a `_dt_spec` structure. I used the same naming convention as is used in SPI: https://github.com/zephyrproject-rtos/zephyr/blob/5113c1418d2b9089be39a7e7dc6601f36e468193/include/zephyr/drivers/spi.h#L339
https://github.com/zephyrproject-rtos/zephyr/blob/5113c1418d2b9089be39a7e7dc6601f36e468193/include/zephyr/drivers/spi.h#L252
See also https://github.com/zephyrproject-rtos/zephyr/pull/43030#discussion_r858913236.